src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -318,13 +427,98 @@ def _prepare_inits(self) -> Set[str]:
   def set_emit_warnings(self, emit_warnings):
     self._builder.info.emit_warnings = emit_warnings
 
+  def _shuffle_original_build_info_into_ipex(self):
+    """Create a "dehydrated" pex file without any of its requirements, and specify that in two *-INFO files.
+
+    See this test case as background:
+    https://github.com/pantsbuild/pex/commit/66aa5ef69e8b65004a548b44edc0b81dfa8ec736#diff-b5b57c9d4fc663f59ecaedb851c13bd8R23
+    """
+    ipex_info = self._builder.info.copy()
+    orig_chroot = self._builder.chroot()
+    # Replace the original builder with a new one, and just pull files from the old chroot.
+    self._builder = PEXBuilder(interpreter=self._builder.interpreter)
+
+    # If the dependencies are very large, this flag is indicated to be useful in RAM-constrained
+    # environments.
+    self._builder.info.always_write_cache = True
+
+    # PTEX-INFO: Contains resolver settings and source files to copy over into the dehydrated .ipex
+    #            file.
+    all_code = []
+    for label in ['source', 'resource']:
+      for src in orig_chroot.get(label):
+        if src in ['__main__.py', PexInfo.PATH]:
+          continue
+        all_code.append(src)
+        self._builder.chroot().copy(os.path.join(orig_chroot.path(), src), src, label=label)
+
+    python_setup = self._python_setup_subsystem
+    python_repos = self._python_repos_subsystem
+
+    resolver_settings = dict(
+      indexes=python_repos.indexes,
+      find_links=list(self._all_find_links),
+      allow_prereleases=bool(python_setup.resolver_allow_prereleases),
+    )
+
+    ptex_info = dict(
+      code=all_code,
+      resolver_settings=resolver_settings,
+      transitive_requirements_by_platform={
```

Comment:
Pex already handles multiplatform. You can kill this field and all the new book-keeping you added to populate it. The requirements needed are simply those in the `ipex_info.requirements`. At runtime the ptex will be on `pex.platforms.Platform.current()` and use that info to pick out the right subset of wheels / hashes from the `ipex_info.distributions` dict.